build/openwisp_base/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM python:3.7-alpine AS BASE
+
+COPY ./openwisp_base/requirements.txt /opt/openwisp/requirements.txt
+RUN apk add --update --no-cache \
+    --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing \
+    gdal-dev geos-dev zlib-dev jpeg-dev openssl-dev libffi-dev gettext gcc && \
+    apk add --update --virtual .build-deps postgresql-dev git build-base linux-headers && \
+    pip install --no-cache-dir -r /opt/openwisp/requirements.txt && \
+    apk del .build-deps && \
+    rm -rf /var/cache/apk/* /root/.cache/pip/*
+
+# TODO: zlib-dev and jpeg-dev maybe .build-deps need to be testing on it.
+# TODO: Only useful when git is not used to install anything
+# FROM BASE AS PIP_INSTALL
+# ENV PYTHONPATH=/install/lib/python3.7/site-packages
+# RUN apk add --update --virtual .build-deps postgresql-dev git
+# RUN pip install asgi_redis service_identity django-redis psycopg2 uwsgi openwisp-network-topology
+# RUN pip install https://github.com/openwisp/django-netjsonconfig/tarball/master
+# RUN pip install https://github.com/openwisp/django-x509/tarball/master
+# RUN pip install https://github.com/openwisp/django-freeradius/tarball/master
+# RUN pip install https://github.com/openwisp/openwisp-radius/tarball/master
+# RUN pip install https://github.com/openwisp/openwisp-users/tarball/master
+# FROM BASE
+# COPY --from=PIP_INSTALL /usr/local /usr/local
```

Comment:
My only suggestion is to include anything else that is common to all containers.  It looks like this may be a candidate
>COPY ./common/ /opt/openwisp/

Reply:
Hi,

Are you taking about: `COPY --from=PIP_INSTALL /usr/local /usr/local` here?
I was testing it to achieve something like Multistage builds [here](https://blog.realkinetic.com/building-minimal-docker-containers-for-python-applications-37d0272c52f3).

(If you mean some other line than the highlighted one or I've misunderstood, please let me know.)

Reply:
No sorry I'm referring to this line in the other Dockerfiles https://github.com/openwisp/docker-openwisp/blob/646147986fb0652a7ca972395ba43d4686b4f03b/build/openwisp_controller/Dockerfile#L5

Reply:
Please can you check again if I understood you correctly. :-)

Reply:
Yes I believe this takes care of it. https://github.com/openwisp/docker-openwisp/blob/fe4f0a3849e5c52b3a522f655e0e5bc822f27c29/build/openwisp_base/Dockerfile#L13